crates/database/src/auction_prices.rs

```diff
@@ -0,0 +1,82 @@
+use {
+    crate::{auction::AuctionId, Address},
+    bigdecimal::BigDecimal,
+    sqlx::PgConnection,
+};
+
+/// External token price for a given auction.
+#[derive(Debug, Clone, PartialEq, sqlx::FromRow)]
+pub struct AuctionPrice {
+    pub auction_id: AuctionId,
+    pub token: Address,
+    pub price: BigDecimal,
+}
+
+pub async fn insert(ex: &mut PgConnection, prices: Vec<AuctionPrice>) -> Result<(), sqlx::Error> {
```

Comment:
I think it would probably be best to use a DB transaction here. Otherwise half of the inserts might fail and we assume everything is in order when looking at the DB.

Reply:
I think you only have to a transaction:
```rust 
pub async fn insert(ex: &mut PgConnection, prices: Vec<AuctionPrice>) -> Result<(), sqlx::Error> {
    let mut ex = ex.begin().await?;
    // existing insert logic
    ex.commit().await
}
```

Reply:
The same pattern can be applied across multiple functions by passing in a `&mut PgTransaction<'_>` to each one. This would be nicer in case we want to insert multiple things in different tables at once. (Which might be needed for the new logic)

Reply:
I also think we should follow the structure of the `database` crate and pass in a `PgTransaction` here. Kind of what we do for other queries:

https://github.com/cowprotocol/services/blob/43a7f9cb9271c47d7c1cce6b2cda72286e106351/crates/database/src/events.rs#L78-L95

Reply:
Fixed.